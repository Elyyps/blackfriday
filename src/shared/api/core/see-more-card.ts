import { Link } from "./link";

export class SeeMoreCard {
  public icon: string;
  public link: Link;
  public title: string;
}
