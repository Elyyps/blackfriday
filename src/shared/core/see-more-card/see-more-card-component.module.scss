.see-more-card {
  min-height: 300px;
  height: 300px;
  max-height: 300px;
  border-radius: 2px;
  @media (max-width: 768px) {
    min-height: 168px;
    height: 168px;
    max-height: 168px;
  }
  min-width: 215px;
  width: 100%;
  background-color: white;
  text-align: left;
  padding: 32px;
  color: black;
  display: flex;
  flex-direction: column;
  height: 88px;
  justify-content: center;
  -webkit-box-sizing: border-box; /* Safari/Chrome, other WebKit */
  -moz-box-sizing: border-box; /* Firefox, other Gecko */
  box-sizing: border-box; /* Opera/IE 8+ */

  &__icon {
    margin-bottom: 16px;
  }
  &__title {
    h4 {
      font-size: 16px;
      margin-bottom: 8px;
    }
  }
  &__content {
    align-items: center;
    justify-content: space-between;
  }

  &--responsive {
    max-height: 166px;
    min-height: 166px;
    padding: 16px;
    width: 100%;
    justify-content: flex-start;
    .see-more-card {
      &__body {
        display: flex;
        align-items: flex-start;
      }

      &__title {
        order: -1;
      }
      &__content {
        display: flex;
        @media (max-width: $breakpoint-small-max) {
          justify-content: space-between;
        }
      }
      &__image {
        order: 2;
        width: 90px;
        height: auto;

        flex: 0 0 90px;
        margin: 0 0 0 10px;
      }
      &__icon {
        margin-left: 8px;
      }

      &__price {
        display: block;
        margin: 0;
        flex: 0 0 50%;
      }

      &__action {
        display: flex;
        align-items: flex-end;
        .button {
          width: 50%;
        }
      }
    }
  }
}
