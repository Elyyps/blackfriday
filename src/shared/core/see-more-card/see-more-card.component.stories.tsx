import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { SeeMoreCardComponent } from "./see-more-card.component";

storiesOf("SeeMoreCard", module)
  .addDecorator(withA11y)
  .add("Basic implementation", () => (
    <SeeMoreCardComponent title="Wij hebben meer winkels in ons overzicht" link={{ title: "Alle winkels", url: "/" }} />
  ));
