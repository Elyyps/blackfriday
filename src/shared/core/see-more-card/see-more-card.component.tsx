import * as React from "react";
import styles from "./see-more-card-component.module.scss";
import { IconComponent } from "@app/prep/modules-prep/core";
import StoreIcon from "@assets/icons/store.svg";
import { Link } from "@app/api/core/link";
import { LinkComponent } from "../link";
import { getArrow } from "@app/constants/icons";

export interface ISeeMoreCardComponentProps {
  link: Link;
  title: string;
  icon: string;
}

const SeeMoreCardComponent = (props: ISeeMoreCardComponentProps) => {
  const { title, link, icon } = props;

  return (
    <div className={styles["see-more-card"]}>
      <div className={styles["see-more-card__icon"]}>
        <IconComponent size="24px" icon={icon} />
      </div>
      <div className={styles["see-more-card__title"]}>{title}</div>
      <LinkComponent
        to={link.url}
        icon={getArrow(false)}
        animated
        variant="tertiary"
      >
        {link.title}
      </LinkComponent>
    </div>
  );
};

export { SeeMoreCardComponent };
