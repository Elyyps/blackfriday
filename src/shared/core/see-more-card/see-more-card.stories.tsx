import React from "react";
import { shallow } from "enzyme";
import { SeeMoreCardComponent } from "./see-more-card.component";

describe("[SeeMoreCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <SeeMoreCardComponent
        title="Wij hebben meer winkels in ons overzicht"
        link={{ title: "Alle winkels", url: "/" }}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
