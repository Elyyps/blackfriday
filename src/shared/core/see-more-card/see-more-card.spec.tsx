import React from "react";
import { SeeMoreCardComponent } from "./see-more-card.component";
import { shallow } from "enzyme";

describe("[SeeMoreCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <SeeMoreCardComponent
        icon="https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0"
        title="Wij hebben meer winkels in ons overzicht"
        buttonLink="/"
        buttonText="Alle winkels"
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
