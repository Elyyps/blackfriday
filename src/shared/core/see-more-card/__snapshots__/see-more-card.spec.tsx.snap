// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SeeMoreCard] should render component without crashing 1`] = `
<ShadowCardComponent
  shadowSize="small"
>
  <div
    className="see-more-card"
  >
    <div
      className="see-more-card__content"
    >
      <div
        className="see-more-card__icon"
      >
        <IconComponent
          icon="https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0"
          size="24px"
        />
      </div>
      <div
        className="see-more-card__title"
      >
        Wij hebben meer winkels in ons overzicht
      </div>
    </div>
    <LinkComponent
      animated={true}
      icon={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "chevron-right.svg",
        }
      }
      to="/"
      variant="tertiary"
    >
      Alle winkels
    </LinkComponent>
  </div>
</ShadowCardComponent>
`;
